modules/http/CMakeLists.txt

```diff
@@ -0,0 +1,37 @@
+option (ENABLE_CURL "Enable curl destination" ON)
+
+set(HTTP_PARSER_SOURCES
+    http-plugin.h
+    http.c
+    http-parser.c
+    http-parser.h
+    http-plugin.c
+    ${CMAKE_CURRENT_BINARY_DIR}/http-grammar.c
+    ${CMAKE_CURRENT_BINARY_DIR}/http-grammar.h
+)
+
+generate_y_from_ym(modules/http/http-grammar)
+
+bison_target(HttpParserGrammar
+    ${CMAKE_CURRENT_BINARY_DIR}/http-grammar.y
+    ${CMAKE_CURRENT_BINARY_DIR}/http-grammar.c
+    COMPILE_FLAGS ${BISON_FLAGS})
+
+find_package(Curl REQUIRED)
+
+if (Curl_FOUND)
+    option(ENABLE_CURL "Enable curl destination" ON)
+else()
+    option(ENABLE_CURL "Enable curl destination" OFF)
+endif()
+
+if (ENABLE_CURL)
+    add_library(curl MODULE ${HTTP_PARSER_SOURCES})
+
```

Comment:
I see that "curl" is being used for the plugin name (I just understood that from your patchset), but I think that's a mistake even in the Makefile, it should be "http". The first incarnation of this plugin was called "curl", then it was renamed to http later on.

Please rename the library to "http".